astropy/time/core.py

```diff
@@ -225,12 +224,19 @@ def __init__(self, val, val2=None, format=None, scale=None,
                                  'one for each time.'
                                  .format(self.location.shape, self.shape))
 
-    def _init_from_vals(self, val, val2, format, scale, copy):
+    def _init_from_vals(self, val, val2, format, scale, copy,
+                        precision=None, in_subfmt=None, out_subfmt=None):
         """
         Set the internal _format, scale, and _time attrs from user
         inputs.  This handles coercion into the correct shapes and
         some basic input validation.
         """
+        if precision is None:
```

Comment:
OK, tricky indeed.  Possibly the only solution is to push it down to the format classes. Anyway, that is getting beyond the scope of this PR.